src/stroke.rs

```diff
@@ -0,0 +1,630 @@
+    pub dash_offset: f64,
+    /// True if the stroke width should be affected by the scale of a
+    /// transform.
+    ///
+    /// Discussion question: does this make sense here?
+    pub scale: bool,
+}
+
+impl Default for Stroke {
+    fn default() -> Self {
+        Self {
+            width: 1.0,
+            join: Join::Round,
+            miter_limit: 4.0,
+            start_cap: Cap::Round,
+            end_cap: Cap::Round,
+            dash_pattern: Default::default(),
+            dash_offset: 0.0,
+            scale: true,
+        }
+    }
+}
+
+impl Stroke {
+    /// Creates a new stroke with the specified width.
+    pub fn new(width: f64) -> Self {
+        Self {
+            width,
+            ..Default::default()
+        }
+    }
+
+    /// Builder method for setting the join style.
+    pub fn with_join(mut self, join: Join) -> Self {
+        self.join = join;
+        self
+    }
+
+    /// Builder method for setting the limit for miter joins.
+    pub fn with_miter_limit(mut self, limit: f64) -> Self {
+        self.miter_limit = limit;
+        self
+    }
+
+    /// Builder method for setting the cap style for the start of the stroke.
+    pub fn with_start_cap(mut self, cap: Cap) -> Self {
+        self.start_cap = cap;
+        self
+    }
+
+    /// Builder method for setting the cap style for the end of the stroke.
+    pub fn with_end_cap(mut self, cap: Cap) -> Self {
+        self.end_cap = cap;
+        self
+    }
+
+    /// Builder method for setting the cap style.
+    pub fn with_caps(mut self, cap: Cap) -> Self {
+        self.start_cap = cap;
+        self.end_cap = cap;
+        self
+    }
+
+    /// Builder method for setting the dashing parameters.
+    pub fn with_dashes<P>(mut self, offset: f64, pattern: P) -> Self
+    where
+        P: IntoIterator,
+        P::Item: Borrow<f64>,
+    {
+        self.dash_offset = offset;
+        self.dash_pattern.clear();
+        self.dash_pattern
+            .extend(pattern.into_iter().map(|dash| *dash.borrow()));
+        self
+    }
+
+    /// Builder method for setting whether or not the stroke should be affected
+    /// by the scale of any applied transform.
+    pub fn with_scale(mut self, yes: bool) -> Self {
+        self.scale = yes;
+        self
+    }
+}
+
+/// Collection of values representing lengths in a dash pattern.
+pub type Dashes = SmallVec<[f64; 4]>;
+
+/// Internal structure used for creating strokes.
+#[derive(Default)]
+struct StrokeCtx {
+    // Probably don't need both output and forward, can just concat
+    output: BezPath,
+    forward_path: BezPath,
+    backward_path: BezPath,
+    start_pt: Point,
+    start_norm: Vec2,
+    start_tan: Vec2,
+    last_pt: Point,
+    last_tan: Vec2,
+    // if hypot < (hypot + dot) * bend_thresh, omit join altogether
+    join_thresh: f64,
+}
+
+/// Expand a stroke into a fill.
+pub fn stroke(path: impl IntoIterator<Item = PathEl>, style: &Stroke, tolerance: f64) -> BezPath {
```

Comment:
Did we want to add an additional options parameter to this? I believe one use was to toggle optimal curve fitting.